spec-main/guest-window-manager-spec.ts

```diff
@@ -229,6 +229,72 @@ describe('webContents.setWindowOpenHandler', () => {
     browserWindow.webContents.executeJavaScript("window.open('https://127.0.0.1')");
     expect(await browserWindow.webContents.executeJavaScript('42')).to.equal(42);
   });
+
+  it('spawns custom browser window if overrideWindowCreationFunction is provided', (done) => {
+    const browserWindowTitle = 'Overriden browser window';
+    browserWindow.webContents.setWindowOpenHandler(() => {
+      return {
+        action: 'allow',
+        overrideWindowCreationFunction: () => {
+          return new BrowserWindow({ title: browserWindowTitle });
+        }
+      };
+    });
+
+    browserWindow.webContents.on('did-create-window', (window) => {
+      expect(window.title).to.equal(browserWindowTitle);
+      done();
```

Comment:
Please use the `async` test style; tests that run expectations in callbacks cause the whole test suite to exit if this test fails.